lib/uplinkc/convert.go

```diff
@@ -14,26 +14,40 @@ func newBucketInfo(bucket *storj.Bucket) C.BucketInfo {
 	return C.BucketInfo{
 		name:         C.CString(bucket.Name),
 		created:      C.int64_t(bucket.Created.Unix()),
-		path_cipher:  C.uint8_t(bucket.PathCipher),
+		path_cipher:  C.Cipher(bucket.PathCipher),
 		segment_size: C.uint64_t(bucket.SegmentsSize),
 
 		encryption_parameters: convertEncryptionParameters(&bucket.EncryptionParameters),
 		redundancy_scheme:     convertRedundancyScheme(&bucket.RedundancyScheme),
 	}
 }
 
+// newObjectInfo returns a C object struct converted from a go object struct.
+func newObjectInfo(object *storj.Object) C.ObjectInfo {
+	return C.ObjectInfo{
+		version:      C.uint32_t(object.Version),
+		bucket:       newBucketInfo(&object.Bucket),
```

Comment:
Do we need to get the bucket info for every object? If the user wants bucket info it can ask it explicitly.